benchmarks/monitoring_log_benchmark.ipynb

```diff
@@ -0,0 +1,227 @@
+    "# Imports\n",
+    "# Offline Plotly, Pandas, Regex and Datetime\n",
+    "import plotly.plotly as py\n",
+    "import plotly.graph_objs as go\n",
+    "from plotly.offline import download_plotlyjs, init_notebook_mode, plot, iplot\n",
+    "import pandas as pd\n",
+    "import re\n",
+    "from datetime import datetime, timedelta\n",
+    "from dateutil import parser\n",
+    "\n",
+    "# Plotly inline\n",
+    "init_notebook_mode(connected=True)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 17,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "# Generate graphs\n",
+    "def parse_logs(filename_pairs_arr, hours=1):\n",
+    "    \"\"\"\"Function to generate plotly graph of CellRanger Monitor Logs with CellRanger std_out Annotations\n",
+    "    Inputs:\n",
+    "        filename_pairs_arr: Array of usage monitoring logs and matching annotation logs in this format-\n",
+    "            [[\"monitroig_log.log\",\"corresponding_std_out.txt\"], [\"monitroig_log2.log\",\"corresponding_std_out2.txt\"]]\n",
+    "        hours: default length of x axis in hours\n",
+    "    Outpus:\n",
+    "        Plotly Graph of Maps: The Legend items are grouped by usage type, disk, core and mem. The title is the VM stats of the run.\n",
+    "                             The CellRanger task names are outputted in the middle of the graph vertically at the time they occured.\n",
+    "                             The xaxis starts at the time of the first task started, and is defaulted to have the range of hours input.\n",
+    "    \"\"\"\n",
+    "    # Collection Arrays\n",
+    "    # Data: Plotly traces of line graphs of usages\n",
+    "    data = []\n",
+    "    # Start times of every task\n",
+    "    earlys = []\n",
+    "    # Cellranger Tasks\n",
+    "    events = []\n",
+    "    # Iterate through filename pairs and populate collection arrays\n",
+    "    for filename_pair in filename_pairs_arr:\n",
+    "        # monitoring log is first file\n",
+    "        monitor = filename_pair[0]\n",
+    "        # std out is second file\n",
+    "        std = filename_pair[1]\n",
+    "        # read std out file\n",
+    "        with open(std) as b:\n",
+    "            lines = b.readlines()\n",
+    "        # parse through the std out file to get major task names\n",
+    "        for line in lines:\n",
+    "            # theres a bunch of meta output that we ignore like copyright info\n",
+    "            # lines that matter have \"[runtime]\" in them, so get them\n",
+    "            if \"[runtime]\" in line:\n",
+    "                # split the line into components we want\n",
+    "                # EXAMPLE LINE WE WANT\n",
+    "                \"\"\"\"\n",
+    "                2018-07-16 21:13:45 [runtime] (ready)           ID.HJCVJBGX5.MAKE_FASTQS_CS.MAKE_FASTQS.PREPARE_SAMPLESHEET\n",
+    "                \"\"\"\n",
+    "                l = line.strip().split()\n",
```

Comment:
Please avoid using one-character variable names unless it's an index or universal convention like `f` for `file` in `with` statements.  (Even if it's index, avoid using `l` as it's very difficult -- and occasionally impossible -- to visually distinguish from the number 1.)

Reply:
Fixed